src/main/java/org/terasology/core/world/generator/trees/RecursiveTreeGeneratorLSystem.java

```diff
@@ -35,45 +37,75 @@
-        Block bark, Block leaf, int depth, AbstractTreeGenerator treeGenerator) {
+    public RecursiveTreeGeneratorLSystem(int maxDepth, float angle, Map<Character, LSystemRule> ruleSet) {
+        this(maxDepth, angle, ruleSet, 3);
+    }
+
+    public void recurse(CoreChunk view, Random rand,
+                        int posX, int posY, int posZ,
+                        float angleOffset,
+                        CharSequenceIterator axiomIterator,
+                        Vector3f position,
+                        Matrix4f rotation,
+                        Block bark,
+                        Block leaf,
+                        int depth,
+                        AbstractTreeGenerator treeGenerator) {
         Matrix4f tempRotation = new Matrix4f();
         while (axiomIterator.hasNext()) {
             char c = axiomIterator.nextChar();
             switch (c) {
                 case 'G':
                 case 'F':
                     // Tree trunk
+                    float radius = TeraMath.fastFloor(thickness / 2);
+                    Vector3i p =
+                            new Vector3i(posX + (int) position.x, posY + (int) position.y, posZ + (int) position.z);
 
-                    treeGenerator.safelySetBlock(view, posX + (int) position.x + 1, posY + (int) position.y, posZ + (int) position.z, bark);
-                    treeGenerator.safelySetBlock(view, posX + (int) position.x - 1, posY + (int) position.y, posZ + (int) position.z, bark);
-                    treeGenerator.safelySetBlock(view, posX + (int) position.x, posY + (int) position.y, posZ + (int) position.z + 1, bark);
-                    treeGenerator.safelySetBlock(view, posX + (int) position.x, posY + (int) position.y, posZ + (int) position.z - 1, bark);
+                    // placing a "cylinder" around 'p' with h=1 and r='radius'
+                    for (int dx = 0; dx <= radius; dx++) {
+                        for (int dz = 0; dz <= radius; dz++) {
+                            if (dx * dx + dz * dz <= radius * radius) {
+                                treeGenerator.safelySetBlock(view, p.x + dx, p.y, p.z + dz, bark);
+                                treeGenerator.safelySetBlock(view, p.x + dx, p.y, p.z - dz, bark);
+                                treeGenerator.safelySetBlock(view, p.x - dx, p.y, p.z + dz, bark);
+                                treeGenerator.safelySetBlock(view, p.x - dx, p.y, p.z - dz, bark);
+                            }
+                        }
+                    }
 
                     // Generate leaves
                     if (depth > 1) {
                         int size = 1;
-
+                        // placing a "sphere" around 'p' with r=~1.5?
+                        //
+                        // □ □ ■ □ □
+                        // □ ■ ■ ■ □
+                        // ■ ■ ⛝ ■ ■
+                        // □ ■ ■ ■ □
+                        // □ □ ■ □ □
```

Comment:
Consider adding such a schema for the cylinder above as well

Reply:
Will do when we get to the point where we're actually writing JavaDoc for this :wink: